modules/_labs/pagination/react/README.md

```diff
@@ -14,82 +74,555 @@ yarn add @workday/canvas-kit-labs-react-pagination
+
+## Models, Hooks, & Utils
+
+### PaginationModel
+
+The `PaginationModel` is the core of the `Pagination` component. That said, if you're using the
+higher-level context API components and following the basic usage guidelines, you shouldn't need to
+know how the model operates to successfully implement the component. But if you have an advanced use
+case or you're feeling curious, this section is for you.
+
+If `Pagination` was stripped of all its markup, attributes, and styling, what would remain is the
+model. The `PaginationModel` is how we describe the state and behavior of the component. You could
+completely swap out the underlying elements, attributes, and styles, and the model would remain the
+same. `PaginationModel` is an object that is composed of two parts: `state` and `events`. The
+model's `state` describes the current state of the component, and `events` are functions that act on
+that state (also called behaviors). Below is an example `Pagination` model:
+
+```tsx
+const paginationModel = {
+  state: {
+    firstPage,
+    currentPage,
+    lastPage,
+    range,
+    rangeSize,
+    rangeMin,
+    rangeMax
+  },
+  events: {
+    jumpToFirst,
+    jumpToLast,
+    stepToNext,
+    stepToPrevious,
+    setCurrentPage,
+    goToPage,
+  };
+};
+
+```
 
-> Determines the label next to the Go To box. Defaults to 'Go To'. Only usable while showGoTo is set
-> to true.
+#### State
 
-#### customLabel?: (from: number, to: number, items: number, itemLabel: string) => string
+The `state` describes the current state of the `Pagination` component. As events are fired, the
+state is updated, and the component will re-render. Below are the `state` keys and descriptions of
+their values:
 
-> A function to build a custom label below the pagination bar.
+- `firstPage` - the page number for the first page
+- `currentPage` - the page number for the current page
+- `lastPage` - the page number for the last page (it can also be used as a total page count)
+- `range` - an array of page numbers included in the visible range
+- `rangeSize` - the size of the visible range
+- `rangeMin` - the first page number in the visible range
+- `rangeMax` - the last page number in the visible range
 
-#### paginationContainerAriaLabel?: string;
+#### Events
 
-> Customizes the aria label for the Pagination container div. Default is 'Pagination'.
+The Model's `events` describe behaviors that act on `state`. Below are the `events` keys and
+descriptions of their values:
 
-#### previousPageAriaLabel?: string;
+- `jumpToFirst` - sets the current page to the first page
+- `jumpToLast` - sets the current page to the last page
+- `stepToNext` - increments the current page by 1
+- `stepToPrevious` - decrements the current page by 1
+- `setCurrentPage` - sets the current page to a given page number (no safeguards)
+- `goToPage` - sets the current page to a given page number\* (with safeguards)
```

Comment:
There's some inconsistency with these event names: `jumpTo*`, `stepTo*`, `goTo*`. Tabs is using the `CursorModel` which has simpler named events like `next`, `first`, `last`, `previous`. How do we want to name these types of events?

Reply:
Perhaps we want to align `Pagination` with `Cursor` since the concept is the same - a list of things has a cursor.

Reply:
Yeah, I went with `jump` and `step` as they correspond to the button names. I'm okay with switching to `first` `last`, but I'm not sure what you would name `goTo` in that paradigm.

Reply:
Updated. Now using `first`, `previous`, `next`, `last`, and `goTo`.